### ISSUE TITLE
New messages in open thread marked as read while app is in the background

### ISSUE THREAD

**Issue by gsbernstein** [GetStream/stream-chat-swift#2826]
## What did you do?
 - configure our app so the unread count on the badge is based on Stream's unread count
 - disable steam push notifications (not sure if this is important)
 - log in to the app and leave a channel open
 - background the app
 - send a new message to that channel from another user outside the app
 - note badge behavior (first expected/actual)
 - force-quit the app without opening it
 - open the app and note the unread counts on the channels (second expected/actual)

## What did you expect to happen?
Badge: Unread count on the badge should increment for any new messages
After force-quit and reopen: channel's unread count should reflect any new messages

## What happened instead?
Badge: Unread count briefly increments, but then resets
After force-quit and reopen: channel unread count is zero, without being viewed

## GetStream Environment
**GetStream Chat version:** 4.38.0 (and going back to at least 4.28.0)
**GetStream Chat frameworks:** StreamChat, StreamChatUI
**iOS version:** 16.4
**Swift version:** swift-driver version: 1.75.2 Apple Swift version 5.8.1 (swiftlang-5.8.0.124.5 clang-1403.0.22.11.100)
**Xcode version:** 14.3.1
**Device:** iPhone 14 Pro simulator

## Additional context
This is our current fix:
```
    override var isLastMessageFullyVisible: Bool {
        guard UIApplication.shared.applicationState == .active else { return false }
        return super.isLastMessageFullyVisible
    }
```
Based on a [fix for a similar issue I was having](https://github.com/GetStream/stream-chat-swift/pull/2544) in the past, you'll probably want to do the actual fix inside `shouldMarkChannelRead`, but that method isn't open so I can't override it.

This fix successfully prevents messages from being marked as read while the app is backgrounded, but as a result the app doesn't always mark them as read once the app is actually brought to foreground. Better than otherwise, though.

## Videos

Actual | Expected
--- | ---
https://github.com/GetStream/stream-chat-swift/assets/6655924/32613450-706f-49a2-a032-5697cb4b03b0|https://github.com/GetStream/stream-chat-swift/assets/6655924/58ae36f5-78e8-428d-99a3-607a221a76b5

Note: we display the app logo when the app is backgrounded as a privacy measure. It's still in memory.

**Comment by polqf** [comment]
Hi @gsbernstein,

Thank you for the report. You are absolutely right, we are marking as read when `ChatChannelConfig.staysConnectedInBackground` is true, and the app is backgrounded with the ChatChannelVC "visible".

I am pushing a fix for it soon, and this should be part of the next planned release (we are shipping 4.39.0 today / Monday, which will not contain it)

**Comment by nuno-vieira** [comment]
Hi @gsbernstein!

This is now available in the new 4.40.0 Release.

Best,
Nuno

**Comment by gsbernstein** [comment]
Great, thanks!
